### PR TITLE
Fix notification diff

### PR DESCRIPTION
**Pull request by ShiCheng-Lu**
this new notification helper gets the list of pending notifications from the list instead of assuming any notifications registered will exist.

Also sdkConfig in swift.

### CODE REVIEW

File: RadarSDK/RadarNotificationHelper.swift [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        let notificationCenter = UNUserNotificationCenter.current()
+        
+        // if notifications is not null, we update the pending notifications, otherwise we only update the registered notifications list
+        if let notifications {
+            // remove all geofence notifications
+            let requests = await notificationCenter.pendingNotificationRequests()
+            if Task.isCancelled {
+                return
+            }
+            let notificationIdentifiersToRemove = requests.compactMap {
```
Comment (lmeier) [inline RadarSDK/RadarNotificationHelper.swift]: Related to registration silently failing, might we wanna also create a list of notifications that we're not touching (aka quietly occupying slots)? At least to log?

**Review by lmeier — COMMENTED** [review]
Would love to see:
- some tests (esp for concurrent `trackOnce` — might need to invest in good mocking of the track response)
- an articulation of our rollout plan
- a manual QA checklist (esp w/ edge cases)

**Review by lmeier — APPROVED** [review]
Be careful w/ the merge, given we change `RadarSDKConfiguration` in your bug fix PR. But LGTM, this is safe to roll out
